rplidar.go

```diff
@@ -227,6 +227,10 @@ func pointFrom(yaw, pitch, distance float64, reflectivity uint8) (r3.Vector, poi
 	pose2 := spatialmath.NewPoseFromPoint(r3.Vector{X: distance, Y: 0, Z: 0})
 	p := spatialmath.Compose(pose1, pose2).Point()
 
+	// Flip the point on the y axis.
```

Comment:
[q] Are you sure this is the appropriate place for the flip and not line 157? Should that angle, defined on line 157, and given to `pointFrom()` be negative? Multiplied by a different number than 90?

Reply:
Spoke with @jeremyrhyde and @nicksanford offline. It did occur to me that this might not be the exact right place for this change  and that we might be doing unnecessary work somewhere else. For example one of the other calculations may be rotating the X coordinate to the wrong place, and then we're flipping it back here. However it seems no one on the SLAM team really understands what those calculations above are doing, so it would take quite a bit of experimentation to figure out the code. Even though we could understand more about the surrounding code, we're confident that this is correct, so since there have been multiple customer complaints about this issue, I think we should go ahead and merge this and then learn more later. I've filed a ticket to do so [here](https://viam.atlassian.net/browse/RSDK-3113). 

Nick also mentioned that flipping the sign of X/Z may be more performant than multiplying it by -1. I ran a quick benchmark and this was true, so I've changed the code.

```
BenchmarkMultiply
BenchmarkMultiply-8   	1000000000	         0.2905 ns/op
BenchmarkNegative
BenchmarkNegative-8   	1000000000	         0.2885 ns/op
```

Also, flipping Z's sign is not strictly necessary, as rplidar only returns 2D data. It's just logically correct that the z axis would also rotate if the x axis were to rotate, so I included it. However since this is a hot code path (gets called for every point in a point cloud) I'm now omitting this change for improved performance.